### ISSUE TITLE
No init_hidden for mdrnn?

### ISSUE THREAD

**Issue by czxttkl** [ctallec/world-models#15]
I do not see hidden layer function initialization in trainmdrnn.py or mdrnn.py. Can you explain if this is correct? I think you would need initialize hidden layers of RNN since your training data may come from many different episodes. Thanks in advance if you can clarify.

**Comment by ctallec** [comment]
Since we are using pytorch default LSTM implementation without specifying h_0 and c_0, both the hidden units and cell units are initialized to 0 before processing any subsequence. This is somehow the best we can do, since we are sampling random subsequences from rollouts, and thus do not keep subsequences ordered (if subsequences were ordered, we could init h_0 and c_0 with the final h and c from the previous subsequence). In practice, my experience is that you gain more with input shuffling than with maintaining hidden states between sequences. Hope this helps.

**Comment by czxttkl** [comment]
Thanks for clarification.
